dynamodb/dynamodb.js

```diff
@@ -0,0 +1,102 @@
+const aws = require('./helper.js'),
+      maxChunks = 25;
+
+let params = { TableName: 'topics' };
+
+const dynamodb = {
+  /**
+   * Delete the DynamoDB table
+   * @return {Promise}
+   */
+  deleteTable() {
+    return aws.deleteTable(params).promise();
+  },
+  /**
+   * Creates and provisions the DynamoDB table with key schema
+   * @return {Promise}
+   */
+  setup() {
+    params.KeySchema = [
+      {
+        AttributeName: 'topic_id',
+        KeyType: 'HASH'
+      }
+    ];
+    params.AttributeDefinitions = [
+      {
+        AttributeName: 'topic_id',
+        AttributeType: 'S'
+      }
+    ],
+    params.StreamSpecification = {
+      StreamEnabled: false
+    },
+    params.ProvisionedThroughput = {
+      ReadCapacityUnits: 1,
+      WriteCapacityUnits: 1
+    }
+
+    return aws.createTable(params).promise();
+  },
+  /**
+   * @param  {array} feeds An array of feeds to populate the DynamoDB table with
+   * @return {Promise}
+   */
+  populate(feeds) {
+    const params = { RequestItems: { topics: [] } };
+    const promises = [];
+
+    feeds = feeds.filter(item => item.rss_link).map(value => {
+      const obj = {
+        PutRequest: {
+          Item: {
+            topic_id: { S: value.rss_link }
+          }
+        }
+      };
+
+      for(const key in value) {
+        obj.PutRequest.Item[key] = (value[key] ? { S: value[key].toString() } : { NULL: true });
+      }
+
+      return obj;
+    });
+
+    const totalCalls = Math.ceil(feeds.length / maxChunks);
+
+    for (let i = 0; i < totalCalls; i++) {
+      const chunk = params;
+      chunk.RequestItems.topics = feeds.slice(i * maxChunks, (i * maxChunks) + maxChunks);
+      promises.push(aws.batchWriteItem(chunk).promise());
+    }
+
+    return Promise.all(promises);
```

Comment:
What happens if you have 5 promises and the third one rejects. Looking at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all it seems to suggest that the first promise rejection will reject all, or maybe I'm mis-understanding and it only returns the first rejection, but waits for all to resolve?

> It rejects with the reason of the first promise that rejects.